### PR TITLE
Alot Mk II

### PR DESCRIPTION
**Pull request by rycee**
@teto Here's a rework of the alot module from #350. I've made my changes in a separate commit. Broadly speaking they are

- Enable alot for an account if notmuch is enabled. I can't recall if I said in an earlier comment to you that there should be an account `alot.enable` option but if I did I believe I was wrong 🙂

- Remove the contact completion option, it wasn't used anywhere and from what I can tell in the Alot documentation is wasn't rich enough to work as-is, it would need, at least, a regexp option as well. I'm not entirely certain how to best define these options so I just removed it for now.

- Remove the `sendMailCommand` function and instead put that functionality in the `alot-accounts.nix` file.

- Rename the `extraHookFile` option to simply `hooks`.

- Add basic support for GPG and signatures.

- Various formatting.

Please have a look and see if it is OK for you.

### CODE REVIEW

**Comment by rycee** [comment]
FWIW I haven't tried this very much but was able to open up mail from notmuch and send mail using msmtp with and without signature and gpg signing.

File: modules/programs/alot-accounts.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        then this is set to
+        <command>msmtpq --account=&lt;name&gt; -t</command>.
+      '';
+    };
+  };
+
+  config = mkIf config.notmuch.enable {
+    alot.sendMailCommand = mkOptionDefault (
+      if config.msmtp.enable
+      then "msmtpq --account=${config.name} -t"
```
Comment (teto) [inline modules/programs/alot-accounts.nix]: while using my astroid and alot modules, I realized it makes more sense to `msmtpq --read-envelope-from --read-recipients`. With multiple accounts, both alot and astroid will let you choose interactively the sender address. When using --read-envelope-from, msmtp will choose the appropriate account whereas msmtpq --account forces always the same account.

**Comment by teto** [comment]
As it's my dev branch, I hadn't pushed the last development but I had tried to "nixify" the configuration as you did in my previous PRs, i.e., having a structured extraConfig https://github.com/teto/home-manager/blob/alot_clean_wip/modules/programs/alot.nix

I am fine with text only extraConfig but the structured one might generate a cleaner config, in case the user wants to override the encrypt_by_default for instance (home-manager can only generate "all" or "none" but "trusted" is another possible value).

Having a non structured extraConfig also ease up quick testing/bootstrap. Like I don't think it's worth adding a "tag" submodule, might be best to just leave it as extraConfig.text.
<nixpkgs>/lib/generators.nix contains functions that can be abused but I dunno what you prefer so I'll let you complete this PR and try to clean up my astroid module (astroid is really a great MUA !)

**Comment by rycee** [comment]
I've added an `extraConfig` section for the account. I looked at the INI format and the possibility of generating a file directly from the attribute set but gave up on that. So I just merged this as is, I've already spent way to much time on this module so will leave any future refinement for somebody else to worry about 🙂

**Pull request merged**